lib/cassia/response_handlers/discover_all_services.rb

```diff
@@ -0,0 +1,35 @@
+module Cassia
+  module ResponseHandlers
+    class DiscoverAllServices
+      def initialize(access_controller, router: , device_mac: )
+        @access_controller = access_controller
+        @router = router
+        @device_mac = device_mac
+      end
+
+      def handle(response)
+        if response.success?
+          handle_success(response)
+        else
+          handle_failure(response)
+        end
+        response.success?
+      end
+
+      private
+
+      def handle_success(response)
+        device = @router.connected_devices.detect {|device| device.mac == @device_mac}
+        device = Device.new(mac: @device_mac) if device.nil?
```

Comment:
sorry I meant we can remove this for now actually, just rely on the `detect` and it being in the `connected_devices` list